packages/main/src/DatePicker.ts

```diff
@@ -822,6 +828,31 @@ class DatePicker extends DateComponentBase implements IFormElement {
 		return !!this._isPickerOpen;
 	}
 
+	/**
+	 * Extracts the picker's calendar mode according to the format
+	 *
+	 * @private
+	 * @method
+	 * @name sap.ui.webc.main.DatePicker#extractCalendarModes
+	 * @returns {CalendarMode} The mode to open the calendar
+	 */
+	extractCalendarMode(): CalendarMode {
+		const format = this.getFormat() as DateFormat & { aFormatArray: Array<{type: string}> };
+		const types = format.aFormatArray.map((settings:{type:string}) => {
```

Comment:
missing spaces after `:`

And why is that needed, aFormatArray is defined as Array of {type: string} above, so settings should be known